llmfoundry/utils/exceptions.py

```diff
@@ -354,3 +356,26 @@ class RunTimeoutError(InternalError):
     def __init__(self, timeout: int) -> None:
         message = f'Run timed out after {timeout} seconds.'
         super().__init__(message, timeout=timeout)
+
+
+class DatasetMissingFileError(UserError):
+    """Error thrown when a dataset cannot find a file."""
+
+    def __init__(
+        self,
+        file_name: str,
+        supported_extensions: Any,  # Run into type error when using List[str]
```

Comment:
what's the type error that you get? We do use `List[str]` in other exceptions, see here https://github.com/mosaicml/llm-foundry/blob/9fbfccc832fa5d365d0b5a5bc83218d2ff1623cd/llmfoundry/utils/exceptions.py#L227 for example. Would be great if we can use `List[str]` here.